src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -16,19 +16,16 @@ namespace System.Net.Security.Tests
 {
     using Configuration = System.Net.Test.Common.Configuration;
 
-    public class SslStreamNetworkStreamTest : IDisposable
+    public class SslStreamNetworkStreamTest
     {
-        private readonly X509Certificate2 _serverCert;
-        private readonly X509Certificate2Collection _serverChain;
+        private static readonly X509Certificate2 _serverCert;
+        private static readonly X509Certificate2Collection _serverChain;
 
-        public SslStreamNetworkStreamTest()
+        static SslStreamNetworkStreamTest()
         {
-            (_serverCert, _serverChain) = TestHelper.GenerateCertificates("localhost", this.GetType().Name);
-        }
-
-        public void Dispose()
-        {
-            TestHelper.CleanupCertificates(this.GetType().Name);
+            string name = System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.Name.ToString();
```

Comment:
Isn't this always `nameof(SslStreamNetworkStreamTest)`?  The cctor is declared there, not up-lifted to each deriving type.

Reply:
yes, I originally experimented with other names but right now I want just the class name since it is static. 
I could use guide since primarily want to avoid collisions with future tests using the helper/cleanup but I think name would be easier to debug.

Reply:
That's fine, I just think that if it's always the string "SslStreamNetworkStreamTest" that using nameof instead of reflection is a lot more clear.

Reply:
agreed. updated.